app/handlers/LastYearPayControllerRequestHandler.scala

```diff
@@ -19,37 +19,49 @@ package handlers
 import java.time.LocalDate
 
 import cats.data.Validated.Valid
-import models.UserAnswers
+import models.{PeriodWithPaymentDate, UserAnswers}
 import models.UserAnswers.AnswerV
 import pages._
 import services.PreviousYearPeriod
+import cats.syntax.apply._
 
 trait LastYearPayControllerRequestHandler extends DataExtractor with PreviousYearPeriod {
 
-  def getPayDatesV(userAnswers: UserAnswers): AnswerV[Seq[LocalDate]] = {
-    import cats.syntax.apply._
+  def getPayDatesV(userAnswers: UserAnswers): AnswerV[Seq[LocalDate]] =
+    (
+      userAnswers.getV(PaymentFrequencyPage),
+      getPeriodsWithPaymentDateV(userAnswers)
+    ).mapN { (frequency, periodsWithPayDates) =>
+      val datesWithDuplicates = periodsWithPayDates.flatMap(p => previousYearPayDate(frequency, p))
```

Comment:
Why not just `periodsWithPayDates.flatMap(previousYearPayDate(frequency, _)).distinct`?

Reply:
where would frequency come from in that instance?

Reply:
@jordanrowe From the above `mapN`?